tardis/visualization/widgets/util.py

```diff
@@ -112,6 +113,107 @@ def create_table_widget(
         column_definitions=column_widths_definitions,
     )
 
+def create_table_widget_shell_info(
+    data, col_widths, table_options=None, changeable_col=None
+):
+    """
+    Create an interactive table widget using Panel's Tabulator, supporting
+    data display, interaction, and optional column name changes.
+
+    Parameters
+    ----------
+    data : pandas.DataFrame
+        Data you want to display in table widget
+    col_widths : list
+        A list containing width of each column of data in order (including
+        the index as 1st column). The width values must be proportions of
+        100 i.e. they must sum to 100.
+    table_options : dict, optional
+        A dictionary specifying configuration options for the Tabulator widget.
+        Overrides the default options where applicable.
+    changeable_col : dict, optional
+        A dictionary specifying the information about column that may change its name when
+        the data updates. It must contain two keys:
+        - :code:`index`: The index of the column in the DataFrame :code:`data`.
+        - :code:`other_names`: A list of possible new names for the column.
+
+    Returns
+    -------
+    panel.widgets.Tabulator
+        An interactive Tabulator table widget displaying the DataFrame.
+
+    Raises
+    ------
+    ValueError
+        If the length of :code:`col_widths` does not match the number of
+        columns + 1 (for the index), or if the column widths do not sum to 100.
+    ValueError
+        If :code:`changeable_col` does not contain both 'index' and 'other_names' keys.
+    """
+    if len(col_widths) != data.shape[1] + 1:
+        raise ValueError(
+            "Size of column widths list do not match with "
+            "number of columns + 1 (index) in dataframe"
+        )
+    if sum(col_widths) != 100:
```

Comment:
Perhaps it would be more useful to accept any numbers and just normalize them to 100